apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPoolImpl.cs

```diff
@@ -205,12 +207,22 @@ private void Push(SessionPoolEntry entry)
                     sw.Start();
                 }
                 //create a new session, blocking or throwing if at the limit.
-                var result = await Key.Client.CreateSessionAsync(new DatabaseName(Key.Project, Key.Instance, Key.Database), cancellationToken).ConfigureAwait(false);
-                if (sw != null)
+                await _sessionCreateThrottle.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
How feasible is it that by the time the semaphore has released, there's actually a suitable pooled session anyway? I suspect it's significantly more complex to wait for "either I can create a session or there's a suitable one" but it's worth considering.